app/Livewire/Project/Database/BackupEdit.php

```
@@ -98,58 +151,171 @@ public function syncData(bool $toModel = false)
+            }
+            $s3Repo->repo_number = $repoNumber;
+            $s3Repo->enabled = true;
+            $s3Repo->s3_storage_id = $this->s3RepoStorageId;
+            $s3Repo->retention_full_type = $this->s3RepoRetentionFullType ?? 'count';
+            $s3Repo->retention_full = $this->s3RepoRetentionFull ?? 2;
+            $s3Repo->retention_diff = $this->s3RepoRetentionDiff ?? 7;
+            $s3Repo->save();
+        } else {
+            $this->backup->pgbackrestRepos()->where('type', 's3')->delete();
+        }
+    }
```

PHPMD (2.15.0)

The method syncPgbackrestRepos() has a Cyclomatic Complexity of 11. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)